### PR TITLE
[mesheryctl] handling default port in mesheryctl

### LINKED ISSUE #8208: [mesheryctl] handling default port in mesheryctl
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

#### Current Behavior

If Meshery server port 9081 is busy it runs on another port soo the request made to endpoint gives error `Error: Get "http://localhost:9081/api/filter?page_size=10000": dial tcp 127.0.0.1:9081: connect: connection refused`
#### Desired Behavior

Mesheryctl should be able to recognize the port on which meshery server is running
#### Environment

- Host OS: Mac Linux Windows
- Meshery Server Version: stable-v
- Meshery Client Version: stable-v
- Platform: Docker Kubernetes

---

#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 📚 [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- 📺 [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- ⌨️ [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- ⌨️ [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by suhail34**
**Notes for Reviewers**

This PR fixes #8208 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: mesheryctl/internal/cli/root/app/offboard.go
Comment (leecalcote) [inline mesheryctl/internal/cli/root/app/offboard.go:60]: Unless we are pervasively referring to `mctlCfg.GetBaseMesheryURL()` as `urlString` in all commands, I think this change hurts readability. 

Unless there is some other reason to use a variable to reference the value of `mctlCfg.GetBaseMesheryURL()`, I would like to see us refer to the function directly. `urlString` strikes me as a reference to an App/Filter location, not as the Meshery Server API location.

**Review by leecalcote — COMMENTED** [review]
Which of these errors captures Short Description, Long Description, Probable Cause and Suggested Remediation of the MeshKit Error function? I'm not seeing these captured, but could be missing it.

File: mesheryctl/internal/cli/root/pattern/view.go
Comment (leecalcote) [inline mesheryctl/internal/cli/root/pattern/view.go:73]: Should this be returning probable cause and suggested remediation, too?

**Comment by suhail34** [comment]
yeah i am working on it

File: mesheryctl/internal/cli/root/app/offboard.go
Comment (suhail34) [reply mesheryctl/internal/cli/root/app/offboard.go:60]: I thought that calling it again and again as it is needed at multiple places would lead into increase in execution time as it is reading the config file so i thought it would be good to call it only once

**Comment by suhail34** [comment]
We can merge this one i will create a separate pr from remaining meshkit errors

File: mesheryctl/internal/cli/root/app/app.go
Comment (abdullah1308) [inline mesheryctl/internal/cli/root/app/app.go:77]: @suhail34 What's the reason for using `log.Error(err)` instead of `return err` ?

File: mesheryctl/internal/cli/root/app/offboard.go
Comment (abdullah1308) [reply mesheryctl/internal/cli/root/app/offboard.go]: @suhail34 Can you please make the change back to calling `mctlCfg.GetBaseMesheryURL()` in the functions. Instead of in the command.

**Review by abdullah1308 — CHANGES_REQUESTED** [review]

File: mesheryctl/internal/cli/root/app/app.go
Comment (suhail34) [reply mesheryctl/internal/cli/root/app/app.go:77]: It returns the probable cause ,remediation, long descrip, short descrip, and code which normally returning doesn't do

**Comment by suhail34** [comment]
I have already removed urlString and made use of mctlCfg.GetBaseMesheryURL()

File: mesheryctl/internal/cli/root/app/app.go
Comment (abdullah1308) [reply mesheryctl/internal/cli/root/app/app.go:77]: @suhail34 can you paste an example  screenshot

File: mesheryctl/pkg/utils/helpers.go
Comment (abdullah1308) [inline mesheryctl/pkg/utils/helpers.go:483]: Can rename `baseUrl` to `mesheryServerUrl`

File: mesheryctl/pkg/utils/helpers.go
Comment (abdullah1308) [inline mesheryctl/pkg/utils/helpers.go:570]: Can rename `baseUrl` to `mesheryServerUrl`

File: mesheryctl/pkg/utils/helpers.go
Comment (abdullah1308) [inline mesheryctl/pkg/utils/helpers.go:588]: Can rename `baseUrl` to `mesheryServerUrl`

**Comment by abdullah1308** [comment]
@suhail34 Please fix the lint checks

**Comment by suhail34** [comment]
when i was running make error in mesheryctl it used give me duplicacy error even when i give error code as "replace_me" but now the problem is solved and when we will merge this it with copy mesheryclt_errors_export.json to docs and will update it

**Comment by suhail34** [comment]
you can try by checking out to my pr

File: mesheryctl/helpers/component_info.json
Comment (leecalcote) [inline mesheryctl/helpers/component_info.json:4]: Is there a reason for this large jump?

Comment (suhail34) [reply mesheryctl/helpers/component_info.json:4]: Yes this error ERRO[0000] component_info.next_error_code '1060' is lower than or equal to highest used code '1079'

Comment (suhail34) [reply mesheryctl/helpers/component_info.json:4]: there are more than 60 error code so it was needed

Comment (leecalcote) [reply mesheryctl/helpers/component_info.json:4]: Gotcha.

**Comment by leecalcote** [comment]
Syncing with master

**Review by leecalcote — CHANGES_REQUESTED** [review]
### Current experience:

```
$ mesheryctl version
      	VERSION    	GITSHA
Client	v0.6.112   	46429a0c2
Server	unavailable	unavailable
Unable to communicate with Meshery server.Get "http://localhost:9081/api/system/version": dial tcp [::1]:9081: connect: connection refused.See https://docs.meshery.io for help getting started with Meshery

Checking for latest version of mesheryctl...

  v0.6.112 is not the latest release. Update to v0.6.113.
```

### Experience from this PR:

```
$ mesheryctl version
      	VERSION    	GITSHA
Client	v0.6.113   	305953b26bf9cd55a3a672dbdf194f93ba903dd4
Server	unavailable	unavailable
ERRO[2023-07-31T18:14:30-05:00] Unable to communicate with Meshery server.Get "http://localhost:9081/api/system/version": dial tcp [::1]:9081: connect: connection refused.See https://docs.meshery.io for help getting started with Meshery  app=mesheryctl code=1079 probable-cause= severity=4 short-description="Unable to communicate with Meshery server" suggested-remediation="See https://docs.meshery.io for help getting started with Meshery"
INFO[2023-07-31T18:14:30-05:00]
Checking for latest version of mesheryctl...  app=mesheryctl
INFO[2023-07-31T18:14:31-05:00]
  v0.6.113 is the latest release.            app=mesheryctl
```

How do we avoid this detraction in UX?

**Comment by suhail34** [comment]
> ### Current experience:
> ```
> $ mesheryctl version
>       	VERSION    	GITSHA
> Client	v0.6.112   	46429a0c2
> Server	unavailable	unavailable
> Unable to communicate with Meshery server.Get "http://localhost:9081/api/system/version": dial tcp [::1]:9081: connect: connection refused.See https://docs.meshery.io for help getting started with Meshery
> 
> Checking for latest version of mesheryctl...
> 
>   v0.6.112 is not the latest release. Update to v0.6.113.
> ```
> 
> ### Experience from this PR:
> ```
> $ mesheryctl version
>       	VERSION    	GITSHA
> Client	v0.6.113   	305953b26bf9cd55a3a672dbdf194f93ba903dd4
> Server	unavailable	unavailable
> ERRO[2023-07-31T18:14:30-05:00] Unable to communicate with Meshery server.Get "http://localhost:9081/api/system/version": dial tcp [::1]:9081: connect: connection refused.See https://docs.meshery.io for help getting started with Meshery  app=mesheryctl code=1079 probable-cause= severity=4 short-description="Unable to communicate with Meshery server" suggested-remediation="See https://docs.meshery.io for help getting started with Meshery"
> INFO[2023-07-31T18:14:30-05:00]
> Checking for latest version of mesheryctl...  app=mesheryctl
> INFO[2023-07-31T18:14:31-05:00]
>   v0.6.113 is the latest release.            app=mesheryctl
> ```
> 
> How do we avoid this detraction in UX? The extra error messages are unwanted and not additive in terms of being helpful to the user.

Is it required that meshery server should running while checking the version

**Comment by suhail34** [comment]
> detraction

I didn't get you

**Comment by suhail34** [comment]
Are you talking about both INFO or ERRO

**Comment by leecalcote** [comment]
@suhail34 in this case, both. I don’t know that this is the case in other circumstances, you’ll have to confirm.

**Comment by suhail34** [comment]
In the server section it says it is unavailable that means meshery server isn't running to get the info about version Soo what do you think it should be warning and other both INFO should be Debug?

**Comment by leecalcote** [comment]
@suhail34 no new messages to the user are needed here. No "ERROR" or "INFO" or timestamp is needed to be shown to the user either. 

This command is just an example. The same statements are true for the UX of all other commands. Timestamps are not needed. Error and Info labels are ok, but not necessary either.

**Comment by Philip-21** [comment]
@suhail34 are you still on this pr

**Comment by suhail34** [comment]
It has a dependent meshkit pr by @nebula-aac once that done this can be merged

**Comment by Philip-21** [comment]
> It has a dependent meshkit pr by @nebula-aac once that done this can be merged

Alright

**Comment by Philip-21** [comment]
@leecalcote everything LGTM, I can make my own changes once this get's merged.

File: mesheryctl/internal/cli/root/app/app.go
Comment (leecalcote) [reply mesheryctl/internal/cli/root/app/app.go]: Is this resolved?

File: mesheryctl/pkg/utils/helpers.go
Comment (leecalcote) [reply mesheryctl/pkg/utils/helpers.go]: Please resolve all open comments that have been addressed.

**Comment by l5io** [comment]
🚀 Preview for commit b049fdae3a7a3e14293e038bf887255f2ed6ac30 at: https://64e43e96a7c7281f15618bc2--meshery-docs-preview.netlify.app

File: docs/_data/discuss/meshery.json
Comment (leecalcote) [inline docs/_data/discuss/meshery.json:1]: Isn't this supposed to be autogenerated post-merge? @alphaX86

**Review by leecalcote — CHANGES_REQUESTED** [review]
@suhail34 will you please separate the slew of welcomed error code updates into a different PR and then offer a write up of what behavior you have changed regarding network ports and mesheryctl?

File: mesheryctl/pkg/utils/helpers.go
Comment (leecalcote) [reply mesheryctl/pkg/utils/helpers.go]: Itemize changes that you've made to incorporate feedback and then resolve the feedback when it has been satisfied (feedback incorporated or feedback rejected), @suhail34

**Pull request closed without merging**
